airbyte-integrations/connectors/source-recharge/unit_tests/integration/streams/test_shop.py

```diff
@@ -0,0 +1,26 @@
+#
+# Copyright (c) 2024 Airbyte, Inc., all rights reserved.
+#
+
+import json
+from unittest import TestCase
+
+import freezegun
+from airbyte_cdk.test.mock_http import HttpMocker, HttpResponse
+from airbyte_cdk.test.mock_http.response_builder import find_template
+
+from ..config import NOW
+from ..request_builder import get_stream_request
+from ..utils import config, read_full_refresh
+
+_STREAM_NAME = "shop"
+
+
+@freezegun.freeze_time(NOW.isoformat())
+class TestFullRefresh(TestCase):
+    @HttpMocker()
+    def test_given_one_page_when_read_then_return_records(self, http_mocker: HttpMocker) -> None:
+        template = find_template("shop", __file__)
+        http_mocker.get(get_stream_request(_STREAM_NAME, "2021-01", False).build(), HttpResponse(json.dumps(template), 200))
```

Comment:
nit: There is one thing I don't like in this test and it is that there is a link between the HTTP request and the config but it is never shown. The link is the `access_token`. Reading the test as it is, it seems like no authentication is needed. Should we update that?

Reply:
The header is matched as for the other tests; only in this case is the API version different. How would you like me to update it?

Reply:
Proposal:

```
    def test_when_read_return_single_record(self, http_mocker: HttpMocker) -> None:
        template = find_template("shop", __file__)
        http_mocker.get(get_stream_request(_STREAM_NAME, "2021-01", False, ACCESS_TOKEN).build(), HttpResponse(json.dumps(template), 200))
        output = read_full_refresh(config().with_access_token(ACCESS_TOKEN), _STREAM_NAME)
        assert len(output.records) == 1
```

This is not only good for documenting the behavior but also prevents this test from breaking if someone change the access token in the config or in the request builder

Reply:
[Config already has `access_token` in it](https://github.com/airbytehq/airbyte/blob/85d2469f3e40eaf476efed220cb465fcf086af5e/airbyte-integrations/connectors/source-recharge/unit_tests/integration/config.py#L22), [request also have the access token in it](https://github.com/airbytehq/airbyte/blob/85d2469f3e40eaf476efed220cb465fcf086af5e/airbyte-integrations/connectors/source-recharge/unit_tests/integration/request_builder.py#L52).

Reply:
Yes, I get that. But the link is only implicit in this test which
* as a reader of this test, I would think there is no authentication involved for recharge
* as a dev, I can change the value of the config and I would break all the tests
Again, this is a nit so I'll let you decide if it is worth fixing

Reply:
As the `access_token` is required for all endpoints and consistently used across all tests, I'll keep the current approach to maintain uniformity and avoid redundancy.

Reply:
Just note: There are ways to maintain uniformity and avoid redundancy that do not rely on losing expressiveness. For example:
```
class XTest(TestCase):
    def setUp(self):
        access_token = "an_access_token"
        self._config = config().with_access_token(access_token)
        self._request = RequestBuilder().with_access_token(access_token)
```

Now, the link between the two is made explicit, is documented as constant between all the tests and is resilient to change on default values for the config or the request.

Reply:
Refactored